prov/efa/src/efa_rma.c

```diff
@@ -56,8 +56,10 @@ static inline ssize_t efa_rma_post_read(struct efa_base_ep *base_ep,
 		ofi_total_iov_len(msg->msg_iov, msg->iov_count),
 		msg->addr, (size_t) msg->context, flags);
 
-	assert(msg->iov_count > 0 &&
-	       msg->iov_count <= base_ep->domain->info->tx_attr->iov_limit);
+	if (OFI_UNLIKELY(msg->iov_count > base_ep->domain->device->ibv_attr.max_sge_rd)) {
```

Comment:
This check happens even for `fi_read`/`fi_write` calls which are guaranteed to only provide a single IOV. If we don't want to iterate and want the check, it should be in `efa_rma_readv`, `efa_rma_readmsg` and the equivalent functions for write and RDM protocol paths

Reply:
The RDM path looks messy

We have asserts like this
https://github.com/ofiwg/libfabric/blob/223d0b604f146bfa73be2cf0ac9867e7307228dc/prov/efa/src/rdm/efa_rdm_ope.c#L1478

But we also copy the `iov_cnt`
https://github.com/ofiwg/libfabric/blob/223d0b604f146bfa73be2cf0ac9867e7307228dc/prov/efa/src/rdm/efa_rdm_rma.c#L66
https://github.com/ofiwg/libfabric/blob/223d0b604f146bfa73be2cf0ac9867e7307228dc/prov/efa/src/rdm/efa_rdm_ope.c#L42

Reply:
The txe and msg can have multi iov, it will finally aggregate all the pkes from it

Reply:
Yeah I saw that. Because we iterate and post a single IOV at a time, we should remove the asserts.

Reply:
I think they should be restricted as it is what we returned for the tx iov limit in fi_getinfo. We can add a check for sure